### PR TITLE
0.1.1 Added source class for GCP BigQuery

### PR DESCRIPTION
**Pull request by mrutunjay-kinagi**
**Name**: Added source class for GCP BigQuery
**About**: Implemented BigQuery integration with local-data-platform
**Description** : This implementation enables local-data-platform to pull data from BigQuery , store it as CSV and Upload CSV data to Iceberg.

**Is your pull request related to a issue? Please tag.** #19

### CODE REVIEW

File: local-data-platform/real_world_use_cases/near_data_lake/config/sample_queries/near_transaction.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from collections import namedtuple
+
+Transaction = namedtuple('Transaction', ['query', 'desc'])
```
Comment (tusharchou) [inline local-data-platform/real_world_use_cases/near_data_lake/config/sample_queries/near_transaction.py:3]: @mrutunjay-kinagi can we please move this to local_data_platform.__init__.py

**Comment by tusharchou** [comment]
This looks great @mrutunjay-kinagi !!!!
@redpheonixx @abhishekkar0603  @brmhastra I feel this PR is a great template for Good First PR. Please also let us know how we could plan better.

File: samples/bigQueryTutorial.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+from pathlib import Path
+import os
+
+"""
+creadentials.json is the path to the credentials file that you downloaded from GCP.
+"""  
+local_gcp_credential_path =   Path(os.getcwd()+'/local-data-platform/samples/credentials.json')
+store = BigQuery(credential_path=local_gcp_credential_path)
+
+df = store.query(NEAR_TRANSACTION_QUERY.query)
```
Comment (redpheonixx) [inline samples/bigQueryTutorial.py:12]: Hi @mrutunjay-kinagi please let me know how you are testing the code. Are you using below script?
https://github.com/tusharchou/local-data-platform/blob/main/local-data-platform/tests/test_gcp_connection.py

Hope this is also updated as I am seeing another function in class BiqQuery doing the same.

Comment (mrutunjay-kinagi) [reply samples/bigQueryTutorial.py:12]: I've made changes to the the big query integration in this PR , I'll update the test in upcoming commits.

File: local-data-platform/real_world_use_cases/near_data_lake/config/sample_queries/near_transaction.py (hunk 1 above)
Comment (mrutunjay-kinagi) [reply local-data-platform/real_world_use_cases/near_data_lake/config/sample_queries/near_transaction.py]: Done

**Comment by mrutunjay-kinagi** [comment]
@tusharchou , @redpheonixx  : Time for you guys to review again !

File: local-data-platform/local_data_platform/pipeline/__init__.py [hunk 3]
```diff
@@ -11,7 +11,7 @@ class Pipeline(Flow):
 
     def __init__(self, config: Config, *args, **kwargs):
         self.config = config
-        self.source = Source(**config.metadata['source'])
-        self.target = Target(**config.metadata['target'])
+        # self.source = Source(**config.metadata['source'])
```
Comment (tusharchou) [inline local-data-platform/local_data_platform/pipeline/__init__.py:14]: @mrutunjay-kinagi what went wrong here?

**Review by tusharchou — APPROVED** [review]
This looks great though let's try to reduce files and increase number of commits next time. Cheers!

File: local-data-platform/local_data_platform/store/source/gcp/__init__.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
+from local_data_platform.store.source import Source
+from pathlib import Path
+import json
+from local_data_platform import Credentials
+from local_data_platform import logger
```
Comment (redpheonixx) [inline local-data-platform/local_data_platform/store/source/gcp/__init__.py:5]: this should be
from local_data_platform.logger import log

**Review by redpheonixx — APPROVED** [review]

**Pull request merged**